Quickstart example doesn't work: [TypeInferenceError] Cannot infer type and shape for node name SkipLayerNorm_0

**Describe the bug**
I tried to run the quick start example given in the README and it failed. I thought it was due to environment or lib mismatch, so I tried it again using the quickstart Colab notebook and I got the same error.


**To Reproduce**
Goto https://colab.research.google.com/github/microsoft/Olive/blob/main/examples/getting_started/olive_quickstart.ipynb
Run all the first three cells, especially until this command:

```bash
olive auto-opt \
    --model_name_or_path HuggingFaceTB/SmolLM2-135M-Instruct \
    --output_path models/smolm2 \
    --device cpu \
    --provider CPUExecutionProvider \
    --use_ort_genai \
    --precision int4 \
    --log_level 1
```
See the exception/stacktrace below:

**Olive logs**
```
The cache for model files in Transformers v4.22.0 has been updated. Migrating your old cache. This is a one-time only operation. You can interrupt this and resume the migration later on by calling `transformers.utils.move_cache()`.
0it [00:00, ?it/s]
Loading HuggingFace model from HuggingFaceTB/SmolLM2-135M-Instruct
[2025-03-18 23:13:27,386] [INFO] [run.py:142:run_engine] Running workflow default_workflow
[2025-03-18 23:13:27,389] [INFO] [cache.py:138:__init__] Using cache directory: /content/.olive-cache/default_workflow
[2025-03-18 23:13:27,393] [INFO] [accelerator_creator.py:229:create_accelerators] Running workflow on accelerator specs: cpu-cpu
[2025-03-18 23:13:27,398] [INFO] [engine.py:223:run] Running Olive on accelerator: cpu-cpu
[2025-03-18 23:13:27,399] [INFO] [engine.py:864:_create_system] Creating target system ...
[2025-03-18 23:13:27,399] [INFO] [engine.py:867:_create_system] Target system created in 0.000200 seconds
[2025-03-18 23:13:27,399] [INFO] [engine.py:879:_create_system] Creating host system ...
[2025-03-18 23:13:27,400] [INFO] [engine.py:882:_create_system] Host system created in 0.000193 seconds
[2025-03-18 23:13:31,763] [INFO] [engine.py:683:_run_pass] Running pass conversion:onnxconversion
2025-03-18 23:13:40.097770: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:477] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1742339620.533408    5921 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1742339620.646583    5921 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-03-18 23:13:41.522236: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
We detected that you are passing `past_key_values` as a tuple and this is deprecated and will be removed in v4.43. Please use an appropriate `Cache` class (https://huggingface.co/docs/transformers/v4.41.3/en/internal/generation_utils#transformers.Cache)
/usr/local/lib/python3.11/dist-packages/transformers/models/llama/modeling_llama.py:95: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if sequence_length != 1:
/usr/local/lib/python3.11/dist-packages/transformers/models/llama/modeling_llama.py:435: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if attn_output.size() != (bsz, self.num_heads, q_len, self.head_dim):
[2025-03-18 23:14:36,604] [INFO] [engine.py:757:_run_pass] Pass conversion:onnxconversion finished in 64.841666 seconds
[2025-03-18 23:14:36,609] [INFO] [engine.py:683:_run_pass] Running pass genai_config_only:modelbuilder
GroupQueryAttention (GQA) is used in this model.
Saving GenAI config in /content/.olive-cache/default_workflow/runs/93614264/models
Saving processing files in /content/.olive-cache/default_workflow/runs/93614264/models for GenAI
[2025-03-18 23:14:37,242] [INFO] [engine.py:757:_run_pass] Pass genai_config_only:modelbuilder finished in 0.633296 seconds
[2025-03-18 23:14:37,246] [INFO] [engine.py:683:_run_pass] Running pass transformer_optimizer:orttransformersoptimization
2025-03-18 23:15:04,562 onnx_model [INFO] - Removed 315 Cast nodes with output type same as input
2025-03-18 23:15:25,698 fusion_base [INFO] - Fused SimplifiedLayerNormalization: 61
2025-03-18 23:16:28,708 fusion_base [INFO] - Fused SkipSimplifiedLayerNormalization: 60
2025-03-18 23:17:33,002 fusion_utils [INFO] - Remove reshape node /model/Reshape_1 since its input shape is same as output: [4]
2025-03-18 23:17:33,002 fusion_utils [INFO] - Remove reshape node /model/Reshape_2 since its input shape is same as output: ['batch_size', 1, 'sequence_length', 'past_sequence_length + sequence_length']
2025-03-18 23:17:33,002 fusion_utils [INFO] - Remove reshape node /model/rotary_emb/Reshape since its input shape is same as output: [3]
2025-03-18 23:17:33,002 fusion_utils [INFO] - Remove reshape node /model/layers.0/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,002 fusion_utils [INFO] - Remove reshape node /model/layers.0/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,003 fusion_utils [INFO] - Remove reshape node /model/layers.1/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,003 fusion_utils [INFO] - Remove reshape node /model/layers.1/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,003 fusion_utils [INFO] - Remove reshape node /model/layers.2/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,003 fusion_utils [INFO] - Remove reshape node /model/layers.2/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,004 fusion_utils [INFO] - Remove reshape node /model/layers.3/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,004 fusion_utils [INFO] - Remove reshape node /model/layers.3/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,004 fusion_utils [INFO] - Remove reshape node /model/layers.4/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,004 fusion_utils [INFO] - Remove reshape node /model/layers.4/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,005 fusion_utils [INFO] - Remove reshape node /model/layers.5/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,005 fusion_utils [INFO] - Remove reshape node /model/layers.5/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,005 fusion_utils [INFO] - Remove reshape node /model/layers.6/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,005 fusion_utils [INFO] - Remove reshape node /model/layers.6/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,005 fusion_utils [INFO] - Remove reshape node /model/layers.7/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,006 fusion_utils [INFO] - Remove reshape node /model/layers.7/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,006 fusion_utils [INFO] - Remove reshape node /model/layers.8/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,006 fusion_utils [INFO] - Remove reshape node /model/layers.8/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,006 fusion_utils [INFO] - Remove reshape node /model/layers.9/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,006 fusion_utils [INFO] - Remove reshape node /model/layers.9/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,007 fusion_utils [INFO] - Remove reshape node /model/layers.10/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,007 fusion_utils [INFO] - Remove reshape node /model/layers.10/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,007 fusion_utils [INFO] - Remove reshape node /model/layers.11/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,007 fusion_utils [INFO] - Remove reshape node /model/layers.11/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,008 fusion_utils [INFO] - Remove reshape node /model/layers.12/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,008 fusion_utils [INFO] - Remove reshape node /model/layers.12/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,008 fusion_utils [INFO] - Remove reshape node /model/layers.13/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,008 fusion_utils [INFO] - Remove reshape node /model/layers.13/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,009 fusion_utils [INFO] - Remove reshape node /model/layers.14/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,009 fusion_utils [INFO] - Remove reshape node /model/layers.14/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,009 fusion_utils [INFO] - Remove reshape node /model/layers.15/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,009 fusion_utils [INFO] - Remove reshape node /model/layers.15/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,010 fusion_utils [INFO] - Remove reshape node /model/layers.16/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,010 fusion_utils [INFO] - Remove reshape node /model/layers.16/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,010 fusion_utils [INFO] - Remove reshape node /model/layers.17/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,010 fusion_utils [INFO] - Remove reshape node /model/layers.17/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,010 fusion_utils [INFO] - Remove reshape node /model/layers.18/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,011 fusion_utils [INFO] - Remove reshape node /model/layers.18/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,011 fusion_utils [INFO] - Remove reshape node /model/layers.19/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,011 fusion_utils [INFO] - Remove reshape node /model/layers.19/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,011 fusion_utils [INFO] - Remove reshape node /model/layers.20/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,012 fusion_utils [INFO] - Remove reshape node /model/layers.20/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,012 fusion_utils [INFO] - Remove reshape node /model/layers.21/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,012 fusion_utils [INFO] - Remove reshape node /model/layers.21/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,012 fusion_utils [INFO] - Remove reshape node /model/layers.22/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,012 fusion_utils [INFO] - Remove reshape node /model/layers.22/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,013 fusion_utils [INFO] - Remove reshape node /model/layers.23/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,013 fusion_utils [INFO] - Remove reshape node /model/layers.23/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,013 fusion_utils [INFO] - Remove reshape node /model/layers.24/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,013 fusion_utils [INFO] - Remove reshape node /model/layers.24/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,014 fusion_utils [INFO] - Remove reshape node /model/layers.25/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,014 fusion_utils [INFO] - Remove reshape node /model/layers.25/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,014 fusion_utils [INFO] - Remove reshape node /model/layers.26/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,014 fusion_utils [INFO] - Remove reshape node /model/layers.26/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,015 fusion_utils [INFO] - Remove reshape node /model/layers.27/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,015 fusion_utils [INFO] - Remove reshape node /model/layers.27/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,015 fusion_utils [INFO] - Remove reshape node /model/layers.28/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,015 fusion_utils [INFO] - Remove reshape node /model/layers.28/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:33,015 fusion_utils [INFO] - Remove reshape node /model/layers.29/self_attn/Reshape_3 since its input shape is same as output: [5]
2025-03-18 23:17:33,015 fusion_utils [INFO] - Remove reshape node /model/layers.29/self_attn/Reshape_5 since its input shape is same as output: [5]
2025-03-18 23:17:36,747 onnx_model [INFO] - Removed 68 nodes
2025-03-18 23:17:36,806 onnx_model_gpt2 [INFO] - postprocess: remove Reshape count: 0
2025-03-18 23:17:36,998 onnx_model_bert [INFO] - opset version: 14
[2025-03-18 23:17:46,735] [INFO] [engine.py:757:_run_pass] Pass transformer_optimizer:orttransformersoptimization finished in 189.489253 seconds
[2025-03-18 23:17:46,738] [INFO] [engine.py:683:_run_pass] Running pass peephole_optimizer:onnxpeepholeoptimizer
[2025-03-18 23:17:50,788] [ERROR] [engine.py:752:_run_pass] Pass run failed.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/dist-packages/olive/engine/engine.py", line 740, in _run_pass
    output_model_config = host.run_pass(p, input_model_config, output_model_path)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/systems/local.py", line 29, in run_pass
    output_model = the_pass.run(model, output_model_path)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/passes/olive_pass.py", line 243, in run
    output_model = self._run_for_config(model, self.config, output_model_path)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/passes/onnx/peephole_optimizer.py", line 268, in _run_for_config
    peephole_optimizer.onnxscript_optimize()
  File "/usr/local/lib/python3.11/dist-packages/olive/passes/onnx/peephole_optimizer.py", line 242, in onnxscript_optimize
    onnxscript.optimizer.optimize(self.model)
  File "/usr/local/lib/python3.11/dist-packages/onnxscript/optimizer/__init__.py", line 35, in optimize
    return legacy_optimizer.optimize(model, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/onnxscript/optimizer/_legacy/_optimizer.py", line 60, in optimize
    model = onnx.shape_inference.infer_shapes(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/onnx/shape_inference.py", line 46, in infer_shapes
    inferred_model_str = C.infer_shapes(
                         ^^^^^^^^^^^^^^^
onnx.onnx_cpp2py_export.shape_inference.InferenceError: [TypeInferenceError] Cannot infer type and shape for node name SkipLayerNorm_0. No opset import for domain com.microsoft optype SkipSimplifiedLayerNormalization
[2025-03-18 23:17:50,795] [WARNING] [engine.py:317:run_accelerator] Failed to run Olive on cpu-cpu.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/dist-packages/olive/engine/engine.py", line 313, in run_accelerator
    output_footprint = self._run_no_search(input_model_config, input_model_id, accelerator_spec, output_dir)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/engine/engine.py", line 356, in _run_no_search
    should_prune, signal, model_ids = self._run_passes(input_model_config, input_model_id, accelerator_spec)
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/engine/engine.py", line 639, in _run_passes
    model_config, model_id = self._run_pass(
                             ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/engine/engine.py", line 740, in _run_pass
    output_model_config = host.run_pass(p, input_model_config, output_model_path)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/systems/local.py", line 29, in run_pass
    output_model = the_pass.run(model, output_model_path)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/passes/olive_pass.py", line 243, in run
    output_model = self._run_for_config(model, self.config, output_model_path)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/olive/passes/onnx/peephole_optimizer.py", line 268, in _run_for_config
    peephole_optimizer.onnxscript_optimize()
  File "/usr/local/lib/python3.11/dist-packages/olive/passes/onnx/peephole_optimizer.py", line 242, in onnxscript_optimize
    onnxscript.optimizer.optimize(self.model)
  File "/usr/local/lib/python3.11/dist-packages/onnxscript/optimizer/__init__.py", line 35, in optimize
    return legacy_optimizer.optimize(model, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/onnxscript/optimizer/_legacy/_optimizer.py", line 60, in optimize
    model = onnx.shape_inference.infer_shapes(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/onnx/shape_inference.py", line 46, in infer_shapes
    inferred_model_str = C.infer_shapes(
                         ^^^^^^^^^^^^^^^
onnx.onnx_cpp2py_export.shape_inference.InferenceError: [TypeInferenceError] Cannot infer type and shape for node name SkipLayerNorm_0. No opset import for domain com.microsoft optype SkipSimplifiedLayerNormalization
[2025-03-18 23:17:50,872] [INFO] [engine.py:241:run] Run history for cpu-cpu:
[2025-03-18 23:17:50,899] [INFO] [engine.py:497:dump_run_history] run history:
+------------+-------------------+-----------------------------+----------------+-----------+
| model_id   | parent_model_id   | from_pass                   |   duration_sec | metrics   |
+============+===================+=============================+================+===========+
| a83bee9c   |                   |                             |                |           |
+------------+-------------------+-----------------------------+----------------+-----------+
| 7956a6f3   | a83bee9c          | onnxconversion              |      64.8417   |           |
+------------+-------------------+-----------------------------+----------------+-----------+
| 93614264   | 7956a6f3          | modelbuilder                |       0.633296 |           |
+------------+-------------------+-----------------------------+----------------+-----------+
| 48bd3a3c   | 93614264          | orttransformersoptimization |     189.489    |           |
+------------+-------------------+-----------------------------+----------------+-----------+
```
**Other information**
All details are in Google Colab notebook linked in the README. 
I had also reproduced the error on Ubuntu 22.04, in a conda environment having Python 3.11.11. 
```
pip list | grep 'torch\|transformer\|olive\|onnx'
olive-ai                 0.8.0
onnx                     1.17.0
onnxruntime              1.21.0
onnxruntime-genai        0.6.0
onnxscript               0.2.2
torch                    2.6.0
torchmetrics             1.6.3
transformers             4.44.2
```

--- a/olive/cli/auto_opt.py
+++ b/olive/cli/auto_opt.py
@@ -428,6 +428,7 @@ TEMPLATE = {
             ("model_builder", {"type": "ModelBuilder", "precision": "fp32"}),
             ("genai_config_only", {"type": "ModelBuilder", "precision": "fp32", "metadata_only": True}),
             # model optimization passes
+            ("peephole_optimizer", {"type": "OnnxPeepholeOptimizer"}),
             # use transformer optimizer for fp16 conversion too
             # opt_level set to 0 to avoid graph transformations done by onnxruntime inference sessions
             # that are incompatible with later passes. opt_level > 0 is optional and can be done during session creation
@@ -435,7 +436,6 @@ TEMPLATE = {
                 "transformer_optimizer",
                 {"type": "OrtTransformersOptimization", "opt_level": 0, "float16": False, "keep_io_types": False},
             ),
-            ("peephole_optimizer", {"type": "OnnxPeepholeOptimizer"}),
             # change io types to fp32
             ("fp16_to_fp32", {"type": "OnnxIODataTypeConverter"}),
             # qnn preparation passes

--- a/olive/passes/onnx/peephole_optimizer.py
+++ b/olive/passes/onnx/peephole_optimizer.py
@@ -238,8 +238,17 @@ class ModelOptimizer:
         except ImportError:
             logger.warning("Please install `onnxscript` to apply more optimization.")
             return
-
-        onnxscript.optimizer.optimize(self.model)
+        try:
+            onnxscript.optimizer.optimize(self.model)
+        except Exception as e:
+            if "TypeInferenceError" in str(e):
+                logger.info(
+                    "onnxscript optimizer failed with %s. Rerunning with shape inference disabled.",
+                    str(e),
+                )
+                onnxscript.optimizer.optimize(self.model, onnx_shape_inference=False)
+            else:
+                raise
 
     def onnxoptimizer_optimize(self):
         try:

